hyperspy/model.py

```diff
@@ -1722,12 +1722,12 @@ def __getitem__(self, slices):
             slices,
             self.isNavigation)
         _signal = self.model.signal._slicer(slices, self.isNavigation)
-        if _signal.metadata.Signal.signal_type == 'EELS':
-            _model = _signal.create_model(
-                auto_background=False,
-                auto_add_edges=False)
-        else:
-            _model = _signal.create_model()
+        # let's not do anything automatically, since then it's difficult to keep
```

Comment:
I agree with the principle. However, to change the API we need to wait for `next_major` which may be a too long wait to get all of this fixes...